src/test/java/com/google/research/bleth/services/DatabaseServiceTest.java

```diff
@@ -0,0 +1,247 @@
+package com.google.research.bleth.services;
+
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.dev.LocalDatastoreService;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.research.bleth.simulator.Agent;
+import com.google.research.bleth.simulator.Board;
+import com.google.research.bleth.simulator.Location;
+import junit.framework.TestCase;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@RunWith(MockitoJUnitRunner.class)
+public final class DatabaseServiceTest extends TestCase {
+
+    private final LocalServiceTestHelper helper =
+            new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig()
+                    .setAutoIdAllocationPolicy(LocalDatastoreService.AutoIdAllocationPolicy.SEQUENTIAL));
+
+    private static final String firstSimulationId = "test-tracing-sim-id-1";
```

Comment:
Constants should be ALL_UPPERCASE.
https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names